### PR TITLE
chore: add export data to migrate tool

### PR DESCRIPTION
**Pull request by shuiyisong**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This pr mainly adds export-data to cli export tool.

### export

e.g
```shell
# export create table
./greptime cli export --addr=127.0.0.1:4001 --output-dir=/tmp/export/ --target=create-table --database="greptime-*"

# export table data
./greptime cli export --addr=127.0.0.1:4001 --output-dir=/tmp/export/ --target=table-data --database="greptime-*"
``` 

result
```
/tmp/export
❯ tree
.
├── greptime-public.sql
├── greptime-thello
│   └── monitor.parquet
├── greptime-thello.sql
└── greptime-thello_copy_from.sql

2 directories, 4 files
```

### import
```shell
# create database forehand

# psql
# create tables
psql -h 127.0.0.1 -p 4003 -d thello -a -f greptime-thello.sql
# copy from data
psql -h 127.0.0.1 -p 4003 -d thello -a -f greptime-thello_copy_from.sql

# mysql
mysql -h 127.0.0.1 -P 4002 public < greptime-public.sql
```

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/cmd/src/cli/export.rs [hunk 1]
```diff
@@ -284,14 +284,64 @@ impl Export {
...
+
+                let sql = format!(
+                    "copy database {} to '{}' with (format='parquet');",
+                    schema,
+                    output_dir.to_str().unwrap()
+                );
+                client
+                    .sql(sql.clone())
+                    .await
+                    .context(RequestDatabaseSnafu { sql })?;
```
Comment (waynexia) [inline src/cmd/src/cli/export.rs:314]: Does this command need to generate a `.sql` file for `COPY FROM`?

**Review by waynexia — APPROVED** [review]
Rest LGTM

File: src/cmd/src/cli/export.rs (hunk 1 above)
Comment (shuiyisong) [reply src/cmd/src/cli/export.rs:315]: @waynexia PTAL

File: src/cmd/src/cli/export.rs [hunk 2]
```diff
@@ -284,14 +284,98 @@ impl Export {
...
+                    })
+                    .collect::<Vec<_>>()
+                    .join("");
+                file.write_all(copy_from_sql.as_bytes())
+                    .await
+                    .context(FileIoSnafu)?;
+
+                info!("finished exporting {catalog}.{schema} copy_from.sql");
+
+                Ok::<(), Error>(())
```
Comment (fengys1996) [inline src/cmd/src/cli/export.rs:351]: nit:
```suggestion
                Ok(())
```

File: src/cmd/src/cli/export.rs [hunk 3]
```diff
@@ -284,14 +284,98 @@ impl Export {
...
+                };
+
+                let copy_from_file =
+                    Path::new(&self.output_dir).join(format!("{catalog}-{schema}_copy_from.sql"));
+                let mut file = File::create(copy_from_file).await.context(FileIoSnafu)?;
+
+                let copy_from_sql = dir_filenames
+                    .into_iter()
+                    .map(|file| {
+                        let file = file.unwrap();
```
Comment (fengys1996) [inline src/cmd/src/cli/export.rs:334]: nit: unwrap safe?

**Review by fengys1996 — APPROVED** [review]
LGTM

File: src/cmd/src/cli/export.rs (hunk 2 above)
Comment (shuiyisong) [reply src/cmd/src/cli/export.rs:351]: Removing this would cause compile error.

File: src/cmd/src/cli/export.rs [hunk 4]
```diff
@@ -284,14 +284,98 @@ impl Export {
...
+                        let filename = file.file_name().into_string().unwrap();
+
+                        format!(
+                            "copy {} from '{}' with (format='parquet');\n",
+                            filename.replace(".parquet", ""),
```
Comment (evenyag) [inline src/cmd/src/cli/export.rs:339]: suggestion(non-blocking): Can we assign this to a variable first? I took a long time to realize that it is the table name.
```rust
let table_name = filename.replace(".parquet", "");
```

File: src/cmd/src/cli/export.rs [hunk 5]
```diff
@@ -284,14 +284,98 @@ impl Export {
...
+                            file.path().to_str().unwrap()
+                        )
+                    })
+                    .collect::<Vec<_>>()
+                    .join("");
```
Comment (evenyag) [inline src/cmd/src/cli/export.rs:344]: Can we collect it into a string?
```suggestion
                    })
                    .collect::<String>();
```

**Pull request merged**